Source/AwsCommonRuntimeKit/io/ClientBootstrap.swift

```diff
@@ -1,33 +1,32 @@
 //  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 //  SPDX-License-Identifier: Apache-2.0.
 import AwsCIo
-import Foundation
 
-public final class ClientBootstrap {
+public class ClientBootstrap {
     let rawValue: UnsafeMutablePointer<aws_client_bootstrap>
-    var enableBlockingShutdown: Bool = false
 
-    public init(eventLoopGroup elg: EventLoopGroup,
-                hostResolver: HostResolver,
-                allocator: Allocator = defaultAllocator,
-                shutdownCallback: ShutdownCallback? = nil) throws {
-        let shutdownCallbackCore = ShutdownCallbackCore(shutdownCallback)
+    public init(clientBootstrapOptions: ClientBootstrapOptions, allocator: Allocator = defaultAllocator) throws {
+        let shutdownCallbackCore = ShutdownCallbackCore(clientBootstrapOptions.shutdownCallback)
         let shutdownOptions = shutdownCallbackCore.getRetainedShutdownOptions()
-        var options = aws_client_bootstrap_options(
-            event_loop_group: elg.rawValue,
-            host_resolver: hostResolver.rawValue,
-            host_resolution_config: hostResolver.config,
-            on_shutdown_complete: shutdownOptions.shutdown_callback_fn,
-            user_data: shutdownOptions.shutdown_callback_user_data
-        )
-        guard let rawValue = aws_client_bootstrap_new(allocator.rawValue, &options) else {
+        guard let rawValue = (clientBootstrapOptions.withCPointer(shutdownOptions: shutdownOptions) { clientBootstrapOptionsPointer in
+            return aws_client_bootstrap_new(allocator.rawValue, clientBootstrapOptionsPointer)
+        }) else {
             shutdownCallbackCore.release()
             throw CommonRunTimeError.crtError(.makeFromLastError())
         }
-
         self.rawValue = rawValue
     }
 
+    public convenience init(eventLoopGroup: EventLoopGroup,
+                            hostResolver: HostResolver,
+                            allocator: Allocator = defaultAllocator,
+                            shutdownCallback: ShutdownCallback? = nil) throws {
+        try self.init(clientBootstrapOptions: ClientBootstrapOptions(eventLoopGroup: eventLoopGroup,
+                                                                     hostResolver: hostResolver,
+                                                                     shutdownCallback: shutdownCallback),
+                      allocator: allocator)
+    }
```

Comment:
We have a couple of options here

- Nuke convenience init and just have init from clientBootstrapOptions
- Nuke clientBootstrapOptions and take elg, hostResolver etc directly like we are doing it in HostResolver.
- Keep both.

 I like both but it is not consistent with other classes.